lib/l10n_utils/__init__.py

```diff
@@ -169,7 +169,9 @@ def render(request, template, context=None, ftl_files=None, activation_files=Non
                 translations.update(ftl_active_locales(af))
             translations = sorted(translations)  # `sorted` returns a list.
         elif l10n:
-            translations = l10n.active_locales
+            translations = l10n.active_locales if not is_root_path_with_no_language_clues(request) else l10n.active_home_locales
+            allowed = settings.PROD_LANGUAGES if not settings.DEV else settings.DEV_LANGUAGES
+            translations = set(translations).intersection(allowed)
```

Comment:
I had to pick a place where to allow only approved locales among all the layers it travels through… but it seemed the least intrusive to just do it here, after getting stuff from either context or l10n instance, and also before adding any extra locales in case the extras are intentionally outside of the allowed range, not filtering them out later. Also, not filtering if the language list comes from other source as CMS translations or activation files (=keeping most of the current logic intact and only making changes that may ever impact the is_root case).